File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    const answer = [];
    const pointObj = {};
    
    for (let i = 0; i < name.length; i++) {
        pointObj[name[i]] = yearning[i]
    }
    
    photo.forEach((item) => {
        let point = 0;
        item.forEach((name) => {
            if (pointObj[name]) point += pointObj[name];
        })
        answer.push(point)
    })
    
    return answer;
}

File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    
    obj = {}
    # object 채우기
    for index, item in enumerate(name):
        obj[item] = yearning[index]
        
    for item in photo:
        tmp = 0
        for target in item:
            if target in obj:
               tmp += obj[target]
        answer.append(tmp)
    return answer